types/serverless/plugins/aws/provider/awsProvider.d.ts

```diff
@@ -457,6 +457,7 @@ declare namespace Aws {
         startingPosition?: number | string | undefined;
         enabled?: boolean | undefined;
         type?: 'dynamodb' | 'kinesis' | undefined;
+        filterPatterns?: object[] | undefined;
```

Comment:
You could go further by writing out types for the 'Filter rule syntax' section of https://docs.aws.amazon.com/lambda/latest/dg/invocation-eventfiltering.html

Something like (just a brief stab at it, probably doesn't work but gives a rough idea):

```ts
type NumericFilter =
  | ["=", number]
  | ["<", number]
  | ["<=", number]
  | [">", number]
  | [">=", number]
type Filter =
  /* Null */
  | null
  /* Empty */
  | ""
  /* String equality */
  | string
  /* Not */
  | { "anything-but": Filter }
  /* Numeric filters */
  | { "numeric": NumericFilter }
  /* Exists filters */
  | { "exists": boolean }
  /* Prefix */
  | { "prefix": string }
type FilterPattern = { [k: string]: FilterPattern | Filter[] }
// ...
filterPatterns?: FilterPattern[] | undefined
```

Reply:
Thank you for your review!
I agree with going with a more specific type, so I will modify it.